### ISSUE TITLE
pgp encryption decryption of file in memory

### ISSUE THREAD

**Issue by vsajip** [vsajip/python-gnupg#156]
**[Original report](https://bitbucket.org/vinay.sajip/python-gnupg/issue/156) by Siddhartha Bose (Bitbucket: [Siddhartha Bose](https://bitbucket.org/Siddhartha Bose), ).**

----------------------------------------

The encryption decryption module takes a output parameter which lets you create a non .gpg file from encrypted file and vice versa for decryption. but this means there will be 2 files at any given point of time. Is there a way encryption and decryption module encryptes and decryptes the file in stdout and then replaces the original file so in disk at any given point of time only 1 file exists.

**Comment by vsajip** [comment]
Please don’t raise issues just to ask questions. The place for asking questions is the [mailing list / discussion group](https://groups.google.com/g/python-gnupg).

**Comment by vsajip** [comment]
**Original changes by Vinay Sajip (Bitbucket: [vinay.sajip](https://bitbucket.org/vinay.sajip), GitHub: [vsajip](https://github.com/vsajip)).**

----------------------------------------

changed state from "new" to "invalid"

**Comment by vsajip** [comment]
**Original comment by Siddhartha Bose (Bitbucket: [Siddhartha Bose](https://bitbucket.org/Siddhartha Bose), ).**

----------------------------------------

Sorry for that @{557058:5393d5ae-3d33-42ba-bea9-4bd1934509bd} , Actually after looking all around for possible solution and then I landed here, We have been using python-gnupg all thanks to you. Recently we have encountered a usecase for which i needed a solution. I will raise the question in the group. Thanks Again
